sklearn/utils/estimator_checks.py

```diff
@@ -569,6 +574,172 @@ def check_estimator_sparse_data(name, estimator_orig):
             raise
 
 
+@contextmanager
+def record_memory_growth():
+    process = psutil.Process()
+    initial_memory = process.memory_info().rss
+    max_memory = [initial_memory]
```

Comment:
```suggestion
    max_memory = initial_memory
```
use `nonlocal` inside `profiler` to set this, rather than using the Python 2 container hack,.

Reply:
TIL 😄